prow/clonerefs/run.go

```diff
@@ -120,6 +124,29 @@ func (o Options) Run() error {
 	return nil
 }
 
+func needsGlobalCookiePath(cookieFile string, refs ...prowapi.Refs) string {
+	if cookieFile == "" || len(refs) == 0 {
+		return ""
+	}
+
+	for _, r := range refs {
+		if !r.SkipSubmodules {
+			return cookieFile
+		}
+	}
+	return ""
+}
+
+// configureGlobalCookiefile ensures git authenticates submodules correctly.
+//
+// Since this is a global setting, we do it once and before running parallel clones.
+func configureGlobalCookiefile(cookiePath string) error {
+	if err := exec.Command("git", "config", "--global", "http.cookiefile", cookiePath).Run(); err != nil {
+		return fmt.Errorf("git config --global http.cookiefile %q: %w", cookiePath, err)
```

Comment:
This won't be included in the text clone log or structured clone record anymore (https://github.com/kubernetes/test-infra/blob/master/prow/pod-utils/clone/types.go) which is unfortunate. Including it would require some changes to support an initialization `Record` that isn't associated with refs.

Reply:
Addressed in https://github.com/kubernetes/test-infra/pull/19231